_drafts/ble-throughput-primer.md

```diff
@@ -0,0 +1,297 @@
+
+`251 bytes / 2500μs = 100.4 kBytes/sec = ~0.803 Mbps`
+
+### LE 2M PHY (BT v5.0)
+
+As part of the 5.0 Bluetooth Core Specification revision, a new feature known as "LE 2M PHY"[^4] was added. As you may recall in the section [above](#ble-radio), we discussed how the BLE Radio is capable of transmitting 1 symbol per μs for a bitrate of 1Mbit/sec. This revision to the Bluetooth Low Energy Physical Layer (PHY), allows for a symbol rate of 2Mbit/sec. This means we can transmit each individual bit in half the time. However, the 150μs IFS is still needed between transmissions. Let's take a look on how this impacts the throughput when sending packets that are using the data packet length extension feature:
+
+![](img/ble-throughput/ble-throughput-dple-and-2m-phy.png)
+
+We can calculate this throughput and see the modification yields almost a _4x_ improvement over the original maximal raw data speed that could be achived with BLE 4.0
+
+`251 bytes / 1400μs = 179.3 kBytes/sec = ~1.434 Mbps`
+
+### LE Coded PHY (BT v5.0)
+
+The "LE Coded PHY" feature, also introduced in the 5.0 Spec [^4] provides a way to extend the range of BLE at the cost of speed. The feature works by encoding a bit across multiple symbols on the radio. This makes the bits being transmitted more resilient to noise. There are two operation modes where either two symbols or eight symbols can be used to encode a single bit. This effectively reduces the radio bitrate to
+500 kbps or 125kbps, respectively.
+
+## Examples
+
+{:.no_toc}
+
+### iPhone SE BLE Throughput Analysis
+
+For iOS 10 and above, most iOS devices support a 185 byte MTU and 7 packets per connection interval. The lowest connection interval which can be negotiated for normal applications is 15ms. Let's take a look at the realistic throughput which can be achieved!
+
+First let's make sure we can transmit 7 packets in 15ms:
+
+`7 * (328μs Data + 150μs + 80μs ACK + 150μs) = 4.956ms`
+
+Now that we know the packets fit, we can take the duty cycle we are achieving and multiply it by the ATT throughput maximums tabulated above.
+
+`Data Rate = (4.956ms / 15ms) * 36.7 kBytes/Sec = 12.2 kBytes/sec = 0.10 Mbps`
+
+{:.no_toc}
+
+### Optimizing throughput Example
+
+Let's figure out if we can improve the throughput for the following application:
+
+- the connection parameter being used is 11.25ms (minimum supported for the device we are using)
+- data packet length extension is being used with 251 byte packets being sent
+- both devices will send as much data they can fit in a connection interval
+- 1 MB PHY is being used
+
+From the examples above, we know it takes 2500μs to exchange a 251 byte data payload. This means we can fit 4 transfers in a 11.25ms connection interval. No data will be sent in the last 1.25ms of the window since it does not fit. This gives us a throughput of:
+
+`(251 * 4) / 11.25 = 89kBytes/sec = 713Mbit/sec`
```

Comment:
You might have mislabeled the units here.

Reply:
hrm, I don't think so. I'll fill this in with some units in the calculation on the left

Reply:
Welp tyler pointed out to me this was indeed wrong, in my head I was reading 713 as .713 :) ... fixed!